### ISSUE TITLE
mec172xevb_assy6906: failed to run tests/arch/arm/irq_advanced_features on v2.7-branch

### ISSUE THREAD

**Issue by hunterhu3000** [zephyrproject-rtos/zephyr#51147]
tests/arch/arm/irq_advanced_features failed to run on mec172xevb_assy6906 on v2.7-branch

 - What target platform are you using? --> mec172xevb_assy6906 
 - What have you tried to diagnose or workaround this issue?  Need a patch from mainline
 - Is this a regression? No

**To Reproduce**
west build -p always -b mec172xevb_assy6906 tests/arch/arm/arm_irq_advanced_features/;west flash

**Expected behavior**
The test should pass

**Impact**
1 test case fail in daily test

**Logs and console output**
```
*** Booting Zephyr OS build zephyr-v2.7.3-28-gf3eb8e821dfb  ***
Running test suite arm_irq_advanced_features
===================================================================
START - test_arm_dynamic_direct_interrupts
Assertion failed at WEST_TOPDIR/zephyr/tests/arch/arm/arm_irq_advanced_features/src/arm_dynamic_direct_interrupts.c:61: test_arm_dynamic_direct_interrupts: (post_flag == 1 is false)
Test flag not set by ISR0
FAIL - test_arm_dynamic_direct_interrupts in 0.19 seconds
===================================================================
START - test_arm_zero_latency_irqs
Available IRQ line: 127
PASS - test_arm_zero_latency_irqs in 0.3 seconds
===================================================================
START - test_arm_irq_target_state
Skipped (TrustZone-M-enabled Cortex-M Mai*** Booting Zephyr OS build zephyr-v2.7.3-28-gf3eb8e821dfb  ***
Running test suite arm_irq_advanced_features
===================================================================
START - test_arm_dynamic_direct_interrupts
Assertion failed at WEST_TOPDIR/zephyr/tests/arch/arm/arm_irq_advanced_features/src/arm_dynamic_direct_interrupts.c:61: test_arm_dynamic_direct_interrupts: (post_flag == 1 is false)
Test flag not set by ISR0
FAIL - test_arm_dynamic_direct_interrupts in 0.19 seconds
===================================================================
START - test_arm_zero_latency_irqs
Available IRQ line: 127
PASS - test_arm_zero_latency_irqs in 0.3 seconds
===================================================================
START - test_arm_irq_target_state
Skipped (TrustZone-M-enabled Cortex-M Mainline only)
PASS - test_arm_irq_target_state in 0.5 seconds
===================================================================
Test suite arm_irq_advanced_features failed.
===================================================================
PROJECT EXECUTION FAILED
*** Booting Zephyr OS build zephyr-v2.7.3-28-gf3eb8e821dfb  ***
Running test suite arm_irq_advanced_features
===================================================================
START - test_arm_dynamic_direct_interrupts
Assertion failed at WEST_TOPDIR/zephyr/tests/arch/arm/arm_irq_advanced_features/src/arm_dynamic_direct_interrupts.c:61: test_arm_dynamic_direct_interrupts: (post_flag == 1 is false)
Test flag not set by ISR0
FAIL - test_arm_dynamic_direct_interrupts in 0.19 seconds
===================================================================
START - test_arm_zero_latency_irqs
Available IRQ line: 127
PASS - test_arm_zero_latency_irqs in 0.3 seconds
===================================================================
START - test_arm_irq_target_state
Skipped (TrustZone-M-enabled Cortex-M Mai*** Booting Zephyr OS build zephyr-v2.7.3-28-gf3eb8e821dfb  ***
Running test suite arm_irq_advanced_features
===================================================================
START - test_arm_dynamic_direct_interrupts
Assertion failed at WEST_TOPDIR/zephyr/tests/arch/arm/arm_irq_advanced_features/src/arm_dynamic_direct_interrupts.c:61: test_arm_dynamic_direct_interrupts: (post_flag == 1 is false)
Test flag not set by ISR0
FAIL - test_arm_dynamic_direct_interrupts in 0.19 seconds
===================================================================
START - test_arm_zero_latency_irqs
Available IRQ line: 127
PASS - test_arm_zero_latency_irqs in 0.3 seconds
===================================================================
START - test_arm_irq_target_state
Skipped (TrustZone-M-enabled Cortex-M Mainline only)
PASS - test_arm_irq_target_state in 0.5 seconds
===================================================================
Test suite arm_irq_advanced_features failed.
===================================================================
PROJECT EXECUTION FAILED
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows) ubuntu 20.04
 - Toolchain (e.g Zephyr SDK, ...) 0.15.0
 - Commit SHA or Version used  a407d69de426d

**Comment by hunterhu3000** [comment]
This issue can be fixed by this patch, if the problem has to be fixed, we need to backport it to v2.7 branch

commit a407d69de426d5e24420cb07a4c482b5ccd39374 (HEAD -> v2.7)
Author: Jay Vasanth <jay.vasanth@microchip.com>
Date:   Thu Dec 16 16:52:07 2021 -0500

    soc arm: MEC172x soc.h - Include custom IRQn_Type

    Fix for issue #41012 to allow compiler to treat
    IRQn_Type to be more than 8-bit. This will ensure NVIC numbers
    more than 127 (required for MEC172x device) will work
    correctly with irq_enable() API

    Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

**Comment by hunterhu3000** [comment]
Fixed by https://github.com/zephyrproject-rtos/zephyr/pull/51106
